src/System.Runtime.Extensions/tests/System/IO/PathTests.cs

```diff
@@ -752,7 +753,10 @@ public static void GetFullPath_Windows_83Paths()
                     string shortName = sb.ToString();
 
                     // Make sure the shortened name expands back to the original one
-                    Assert.Equal(tempFilePath, Path.GetFullPath(shortName));
+                    // Sometimes shortening or GetFullPath is changing the casing of "temp" on some test machines: normalize both sides
+                    tempFilePath = Regex.Replace(tempFilePath, @"\temp\\", @"\TEMP\\",  RegexOptions.IgnoreCase);
```

Comment:
I'm being cautious because I can't reproduce the problem and I don't know that it wouldn't be the other way around on some other machine. It's probably some silly thing like a TEMP environment variable set or some different  registry key.